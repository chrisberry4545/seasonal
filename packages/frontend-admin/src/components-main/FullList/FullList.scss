.c-full-list {
  position: relative;
}

.c-full-list__item {
  margin-bottom: 1rem;
}

.c-full-list__item-name {
  display: block;
}

.c-full-list-item__options {
  display: flex;
}

.c-full-list-item__edit {
  padding-left: 0;
}
