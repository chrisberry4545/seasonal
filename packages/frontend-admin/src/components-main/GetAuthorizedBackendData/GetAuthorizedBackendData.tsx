import React, { FC, useState, useEffect } from 'react';

export interface IGetAuthorizedBackendDataProps<T> {
  items: T;
  reload?: () => void;
  updateMethod?: (item: T) => Promise<T>;
}

interface IGetAuthorizedBackendDataState<T> {
  isLoading: boolean;
  items: T | null;
  error: string | null;
}

export function GetAuthorizedBackendData<T>(
  InnerComponent: FC<IGetAuthorizedBackendDataProps<T>>,
  requestDataMethod: () => Promise<T>,
  updateMethod?: (item: T) => Promise<T>
): FC<{}> {
  return () => {
    const [state, setState] = useState<IGetAuthorizedBackendDataState<T>>({
      error: null,
      isLoading: true,
      items: null
    });

    const requestNewData = () => {
      requestDataMethod()
        .then((items) => setState({
          error: null,
          isLoading: false,
          items
        })).catch((error) => setState({
          error: error.message,
          isLoading: false,
          items: null
        }));
    };

    useEffect(requestNewData, []);

    return (
      <div>
        {
          state.isLoading
            ? <div>Loading...</div>
            : <div>
              {
                !state.error && state.items
                  ? <InnerComponent
                      items={state.items}
                      reload={requestNewData}
                      updateMethod={updateMethod}
                    />
                  : <div>{state.error}</div>
              }
            </div>
        }
      </div>
    );
  };
}
