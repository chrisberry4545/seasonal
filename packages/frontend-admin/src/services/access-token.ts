import { getStorage } from '../consts';

export const setAccessToken = (token: string) => {
  getStorage().accessToken = token;
};

export const getAccessTokenHeaders = (): HeadersInit => {
  const token = getStorage().accessToken;
  return token
  ? {
    Authorization: `Bearer ${token}`
  }
  : {};
};
