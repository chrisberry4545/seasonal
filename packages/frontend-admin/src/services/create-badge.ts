import { IBadge } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { BADGE_URL } from '../config';

export const createBadge = async (badge: IBadge): Promise<IBadge> =>
  makeAuthorizedRequest<IBadge>(BADGE_URL, {
    body: JSON.stringify(badge),
    method: 'POST'
  });
