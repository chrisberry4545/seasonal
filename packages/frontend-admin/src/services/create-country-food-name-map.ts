import { ICountryFoodNameMap } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_FOOD_NAME_MAP_URL } from '../config';

export const createCountryFoodNameMap = async (
  country: ICountryFoodNameMap
): Promise<ICountryFoodNameMap> =>
  makeAuthorizedRequest<ICountryFoodNameMap>(COUNTRY_FOOD_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
