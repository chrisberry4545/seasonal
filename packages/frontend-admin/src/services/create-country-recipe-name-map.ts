import { ICountryRecipeNameMap } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_RECIPE_NAME_MAP_URL } from '../config';

export const createCountryRecipeNameMap = async (
  country: ICountryRecipeNameMap
): Promise<ICountryRecipeNameMap> =>
  makeAuthorizedRequest<ICountryRecipeNameMap>(COUNTRY_RECIPE_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
