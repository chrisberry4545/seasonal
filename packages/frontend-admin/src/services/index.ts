export * from './access-token';

export * from './get-all-countries';

export * from './get-all-food';

export * from './get-all-recipes';

export * from './get-single-recipe';

export * from './login-request';

export * from './make-authorized-request';

export * from './update-recipe';
