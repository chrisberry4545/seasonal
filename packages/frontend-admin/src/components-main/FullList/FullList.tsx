import React, { useState } from 'react';
import { FormLayout } from '../../components-layouts';
import { TextHeadingMedium, TextSmall, TextMedium, BareButton } from '@chrisb-dev/seasonal-shared-frontend-components';
import { Redirect } from 'react-router-dom';
import { DeleteItemButton } from '../DeleteItemButton/DeleteItemButton';
import './FullList.scss';

export interface IFullListProps<T> {
  title: string;
  items: T[] | undefined;
  getItemName: (item: T) => string;
  getItemEditUrl: (item: T) => string;
  deleteItemFunc: (item: T) => Promise<T>;
}

export const FullList = <T extends {}>({
  title,
  items,
  getItemName,
  getItemEditUrl,
  deleteItemFunc
}: IFullListProps<T>) => {
  const [goToEditLink, setGoToEditLink] = useState<null | string>(null);

  if (goToEditLink) {
    return <Redirect to={goToEditLink} push={true} />;
  }

  return <FormLayout>
    <TextHeadingMedium className='c-full-list__heading'>{title}</TextHeadingMedium>
    {
      items && items.map((item) =>
        <div className='c-full-list__item' key={getItemEditUrl(item)}>
          <TextMedium className='c-full-list__item-name'>{getItemName(item)}</TextMedium>
          <div className='c-full-list-item__options'>
            <BareButton className='c-full-list-item__edit'
              onClick={() => setGoToEditLink(getItemEditUrl(item))}>
              <TextSmall>
                Edit
              </TextSmall>
            </BareButton>
            <DeleteItemButton deleteItem={() => deleteItemFunc(item)} />
          </div>
        </div>
      )
    }
  </FormLayout>;
};
