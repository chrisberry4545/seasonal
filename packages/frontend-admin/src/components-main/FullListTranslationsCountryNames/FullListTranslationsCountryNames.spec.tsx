import React from 'react';
import {
  FullListTranslationsCountryNames,
  ITranslationsCountryNameViewItem
} from './FullListTranslationsCountryNames';
import { ShallowWrapper, shallow } from 'enzyme';
import { GetAuthorizedBackendData } from '../GetAuthorizedBackendData/GetAuthorizedBackendData';
import { ICountry } from '@chrisb-dev/seasonal-shared-models';
import { FullList } from '../FullList/FullList';
import * as services from '../../services';

jest.mock('../FullList/FullList', () => ({
  FullList: () => 'FullList'
}));
jest.mock('../../config', () => ({
  ROUTES: {
    EDIT: 'edit',
    TRANSLATIONS_COUNTRY_NAME: 'translations'
  }
}));
jest.mock('../../services', () => ({
  deleteTranslationsCountryName: jest.fn(),
  getAllCountries: jest.fn(),
  getAllTranslationsCountryNames: jest.fn()
}));
jest.mock('../GetAuthorizedBackendData/GetAuthorizedBackendData', () => ({
  GetAuthorizedBackendData: () => 'GetAuthorizedBackendData'
}));

describe('FullListTranslationsCountryNames', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() =>
    wrapper = shallow(
      <FullListTranslationsCountryNames />
    )
  );

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  describe('the inner component', () => {
    let innerWrapper: ShallowWrapper;
    let mockReload: jest.Mock;
    let mockDelete: jest.SpyInstance;
    const item = {
      countryId: 'c1',
      countryName: 'country1',
      id: 'id',
      languages: ['en-GB'],
      name: 'name'
    } as ITranslationsCountryNameViewItem;
    const getFullListProps = () => innerWrapper.find(FullList).first().props();

    beforeEach(async () => {
      mockReload = jest.fn();
      mockDelete = jest.spyOn(services, 'deleteTranslationsCountryName')
        .mockResolvedValue(item);
      jest.spyOn(services, 'getAllTranslationsCountryNames')
        .mockResolvedValue([item]);
      jest.spyOn(services, 'getAllCountries')
        .mockResolvedValue([{
          id: 'c1', name: 'country1'
        }] as ICountry[]);
      mockDelete.mockClear();
      const {
        InnerComponent,
        requestDataMethod
      } = wrapper.find(GetAuthorizedBackendData).first().props();
      innerWrapper = shallow(
        <InnerComponent items={await requestDataMethod()} reload={mockReload} />
      );

    });

    test('renders correctly', () => expect(innerWrapper).toMatchSnapshot());

    test('getItemId returns the items id', () =>
      expect(getFullListProps().getItemId(item)).toBe('id'));

    test('getItemName returns the items name', () =>
      expect(getFullListProps().getItemName(item)).toBe('country1 - en-GB - name'));

    test('getItemEditUrl returns the expected url', () =>
      expect(getFullListProps().getItemEditUrl(item)).toBe(
        'translations/edit/id'
      ));

    describe('when delete is pressed', () => {
      beforeEach(async () => await getFullListProps().deleteItemFunc(item));

      test('calls delete', () =>
        expect(mockDelete).toHaveBeenCalledWith(item.id));

      test('calls reload', () => expect(mockReload).toHaveBeenCalled());

    });
  });

});
