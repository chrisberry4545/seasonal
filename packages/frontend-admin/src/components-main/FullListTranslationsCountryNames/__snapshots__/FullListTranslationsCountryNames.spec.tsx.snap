// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullListTranslationsCountryNames renders correctly 1`] = `
<GetAuthorizedBackendData
  InnerComponent={[Function]}
  requestDataMethod={[Function]}
/>
`;

exports[`FullListTranslationsCountryNames the inner component renders correctly 1`] = `
<FullList
  deleteItemFunc={[Function]}
  getItemEditUrl={[Function]}
  getItemId={[Function]}
  getItemName={[Function]}
  items={
    Array [
      Object {
        "countryId": "c1",
        "countryName": "country1",
        "id": "id",
        "languages": Array [
          "en-GB",
        ],
        "name": "name",
      },
    ]
  }
  title="Translations Country Names"
/>
`;
